src/command/command.rs

```diff
@@ -64,6 +66,35 @@ impl Command {
                 }
                 Ok((Command::Command, 2))
             }
+            _ if cmd == "GET" => {
+                if length != 2 {
+                    return Err(CommandError::Malformed);
+                }
+                let key = match &tokens[2] {
+                    Token::SimpleString(s) => s.bytes().collect(),
+                    Token::BulkString(Some(s)) => s.clone(),
+                    _ => return Err(CommandError::Malformed),
+                };
+
+                Ok((Command::Get(key), 3))
+            }
+            _ if cmd == "SET" => {
+                if length != 3 {
+                    return Err(CommandError::Malformed);
+                }
+                let key = match &tokens[2] {
+                    Token::SimpleString(s) => s.bytes().collect(),
+                    Token::BulkString(Some(s)) => s.clone(),
+                    _ => return Err(CommandError::Malformed),
+                };
+                let value = match &tokens[3] {
+                    Token::SimpleString(s) => s.bytes().collect(),
+                    Token::BulkString(Some(s)) => s.clone(),
+                    _ => return Err(CommandError::Malformed),
+                };
+
+                Ok((Command::Set(key, value), 4))
```

Comment:
It's not evident to me what the second value of your tuple here is. Changing the return type (maybe to a struct that encapsulates `Command`?) might be a task for a future refactor.

Reply:
Also, using constants instead of hard-coded integers improves readability. Even if it was obvious what the second value was, it might not be evident why that value should be 4 in this case.

Reply:
Added constants.

I'm going to think on what the best way to communicate the return type here is. I'm leaving off wrapping it in a struct but that's an option I'll consider down the road. The `usize` returned is how many of the passed-in tokens were used in parsing the Command.

Ultimately I'm probably going to change this to operate on a stream, so if you have an iterable of Tokens you get an iterable of Commands, which would clean up that issue considerably. So I don't want to invest _too_ much into making this particular return type clearer when I might throw it away :smile: